backend/routes/application/application.go

```diff
@@ -0,0 +1,97 @@
+package routes
+
+/*
+/api/v1/application
+GET: Retrieve a list of applications with the most recently updated statuses (or another subset, via query params)
+
+/api/v1/application/(id)
+GET: Retrieve details on a particular application
+
+ToDos (in future): Write Operations (POST, PUT, DELETE)
+*/
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"time"
+
+	restful "github.com/emicklei/go-restful/v3"
+)
+
+// Creating a REST layer for application - ApplicationResource
+
+type Application struct {
+	Id, Name string
+}
+
+type ApplicationResource struct {
+	applications map[string]Application
+}
+
+// Creating a webservice for application endpoints
+func (a ApplicationResource) Register(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/api/v1/application").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("/{application-id}").To(a.findApplication))
+	ws.Route(ws.GET("").To(a.recentApplication))
+	container.Add(ws)
+}
+
+// GET Retrieve a list of applications with the most recently updated statuses
+func (a ApplicationResource) recentApplication(request *restful.Request, response *restful.Response) {
+	list := []Application{}
+	for _, each := range a.applications {
+		list = append(list, each)
+	}
+
+	err := response.WriteEntity(list)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// GET info of operations depening upon the id
+func (a ApplicationResource) findApplication(request *restful.Request, response *restful.Response) {
+	id := request.PathParameter("application-id")
+	app := a.applications[id]
+	if len(app.Id) == 0 {
+		response.AddHeader("Content-Type", "text/plain")
+		err := response.WriteErrorString(http.StatusNotFound, "Application not found!")
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		err := response.WriteEntity(app)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// Add function to start up the server, running against dedicated port
+// Usage of CurlyRouter is done because of the efficiency while using wildcards and expressions
+func RunRestfulCurlyRouterServer() {
+	wsContainer := restful.NewContainer()
+	wsContainer.Router(restful.CurlyRouter{})
```

Comment:
I think we should update the startup logic such that all of the resources should be initialized by, and started from, a single function, for example:
```go
func RunRestfulCurlyRouterServer() {
	wsContainer := restful.NewContainer()
	wsContainer.Router(restful.CurlyRouter{})
	
	// Register first resource
	a.Register(ApplicationResource{map[string]Application{}})

	// Register second resource
	o := OperationResource{map[string]Operation{}}
	o.Register(wsContainer)
	
	// Register nth resource
	// (...)

	
	log.Print("The server is up, and listening to port 8090 on your host.")
	server := &http.Server{Addr: ":8090", Handler: wsContainer}
	log.Fatal(server.ListenAndServe())
}
```	
Eg rather than having each resource have its own server listen function.

Then, this function should be called from `main.go` at the root of the application, so that it can serve requests for all API routes on startup.

Reply:
Absolutely, by having all at one place (say main.go) it will synchonise as we will be having one server to listen to! I'll make the necessary changes! :)